src/main/java/com/sciamus/contractanalyzer/domain/checks/queues/kafka/KafkaTransformationCheck.java

```diff
@@ -0,0 +1,151 @@
+package com.sciamus.contractanalyzer.domain.checks.queues.kafka;
+
+import com.sciamus.contractanalyzer.domain.checks.queues.kafka.config.KafkaConsumFactory;
+import com.sciamus.contractanalyzer.domain.checks.queues.kafka.config.KafkaProducFactory;
+import com.sciamus.contractanalyzer.domain.checks.reports.Report;
+import com.sciamus.contractanalyzer.domain.checks.reports.ReportBuilder;
+import com.sciamus.contractanalyzer.domain.checks.reports.ReportResults;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class KafkaTransformationCheck implements KafkaCheck {
+    final String name = "KafkaTransformationCheck";
+
+    private final KafkaProducFactory kafkaProducFactory;
+    private final KafkaConsumFactory kafkaConsumFactory;
+    private ReportBuilder reportBuilder = new ReportBuilder();
+
+    public KafkaTransformationCheck(KafkaProducFactory kafkaProducFactory, KafkaConsumFactory kafkaConsumFactory) {
+        this.kafkaProducFactory = kafkaProducFactory;
+        this.kafkaConsumFactory = kafkaConsumFactory;
+    }
+
+    @Override
+    public Report run(String incomingTopic, String outgoingTopic, String host, String port) {
+        //given:
+        KafkaTemplate<String, String> producer = Producer(host, port);
+        Consumer<String, String> consumer = Consumer(incomingTopic, host, port);
+        String checkUniqueIdentifier = getCheckUniqueIdentifier();
+        List<Integer> randomIntegers = generateRandomIntegers();
+        String outsideProcessorMessage = convertRandomIntegersToMessage(randomIntegers);
+
+        //when:
+        sendMessageToOutsideSystem(outgoingTopic, checkUniqueIdentifier, producer, outsideProcessorMessage);
+        Integer expectedAnswer = getExpectedAnswer(randomIntegers);
+
+        Iterable<ConsumerRecord<String, String>> recordsFromOutsideSystem = waitForRecordFromOutsideSystem(incomingTopic, consumer);
+        Option<Integer> actualAnswer = checkIfReceivedRecordsHaveUniqueIdentifier(recordsFromOutsideSystem);
+
+        //then:
+        setUpReportBuilder(reportBuilder);
+
+        if (expectedAnswer.equals(actualAnswer.getOrNull()))
```

Comment:
## notice

lack of braces

Reply:
## notice

I think there is a better way of handling Option... How about transforming it into a result with a previous filtering?